mods/ra/maps/allies-11a/allies11a.lua

```diff
@@ -0,0 +1,203 @@
+}
+
+mcv2Reinforcements =
+{
+	actors = { "mcv" },
+	entryPath = { WP0.Location, WP30.Location }
+}
+
+sea1Reinforcements =
+{
+	actors = { "pt", "pt", "dd", "dd" },
+	entryPath = { WP28.Location }
+}
+
+sea2Reinforcements =
+{
+	actors = { "pt", "pt", "dd", "dd" },
+	entryPath = { WP58.Location }
+}
+
+sea3Reinforcements =
+{
+	actors = { "ca" },
+	entryPath = { WP28.Location }
+}
+
+sea4Reinforcements =
+{
+	actors = { "ca" },
+	entryPath = { WP58.Location }
+}
+
+sea3sea4Units = {}
+
+sea1PatrolPath = { WP27.Location, WP52.Location, WP53.Location, WP54.Location, WP55.Location, WP56.Location, WP57.Location }
+sea2PatrolPath = { WP59.Location, WP60.Location, WP61.Location, WP62.Location, WP63.Location, WP64.Location, WP65.Location }
+
+mmthPatrolPath = { WP2.Location, WP3.Location, WP4.Location, WP5.Location, WP6.Location, WP7.Location, WP8.Location, WP9.Location, WP10.Location, WP11.Location }
+
+edgeOfRiverTriggerActivator = 
+{ 
+	CPos.New(50,1), CPos.New(51,1), CPos.New(52,1), CPos.New(53,1), CPos.New(54,1), CPos.New(55,1),
+	CPos.New(56,1), CPos.New(57,1), CPos.New(58,1), CPos.New(59,1), CPos.New(60,1), CPos.New(61,1),
+	CPos.New(62,1), CPos.New(63,1), CPos.New(64,1), CPos.New(65,1), CPos.New(66,1), CPos.New(67,1),
+	CPos.New(68,1), CPos.New(69,1), CPos.New(70,1), CPos.New(71,1), CPos.New(72,1), CPos.New(73,1),
+	CPos.New(74,1), CPos.New(75,1), CPos.New(76,1), CPos.New(77,1), CPos.New(78,1), CPos.New(79,1),
+	CPos.New(80,1), CPos.New(81,1), CPos.New(82,1), CPos.New(83,1), CPos.New(84,1), CPos.New(85,1),
+	CPos.New(50,2), CPos.New(51,2), CPos.New(52,2), CPos.New(53,2), CPos.New(54,2), CPos.New(55,2),
+	CPos.New(56,2), CPos.New(57,2), CPos.New(58,2), CPos.New(59,2), CPos.New(60,2), CPos.New(61,2),
+	CPos.New(62,2), CPos.New(63,2), CPos.New(64,2), CPos.New(65,2), CPos.New(66,2), CPos.New(67,2),
+	CPos.New(68,2), CPos.New(69,2), CPos.New(70,2), CPos.New(71,2), CPos.New(72,2), CPos.New(73,2),
+	CPos.New(74,2), CPos.New(75,2), CPos.New(76,2), CPos.New(77,2), CPos.New(78,2), CPos.New(79,2),
+	CPos.New(80,2), CPos.New(81,2), CPos.New(82,2), CPos.New(83,2), CPos.New(84,2), CPos.New(85,2)
+}
+
+
+TimerTicks = DateTime.Minutes(120)
+
+MissionAccomplished = function()
+	Media.PlaySpeechNotification(Greece, "MissionAccomplished")
+end
+
+MissionFailed = function()
+	Media.PlaySpeechNotification(Greece, "MissionFailed")
+end
+
+InitialSovietPatrols = function()
+	mmth1.Patrol(mmthPatrolPath, true, DateTime.Seconds(12))
+	mmth2.Patrol(mmthPatrolPath, true, DateTime.Seconds(12))
+end
+
+InitialAlliedReinforcements = function()
+	Trigger.AfterDelay(DateTime.Seconds(1), function()
+		Media.PlaySpeechNotification(Greece, "ReinforcementsArrived")
+		Reinforcements.Reinforce(Greece, mcv1Reinforcements.actors, mcv1Reinforcements.entryPath)
+	end)
+	Trigger.AfterDelay(DateTime.Seconds(1), function()
+		Reinforcements.Reinforce(Greece, mcv2Reinforcements.actors, mcv2Reinforcements.entryPath)
+	end)
+end
+
+timerExpiredSendNavy = function()
```

Comment:
We'll want to send the fleet if the player destroys all opposition before the timer runs out. Just add a check so it can't send the fleet twice.

Reply:
Add something like this
`if SentNavy then
		return
	end`

Reply:
I meant I don't see any logic to activate `TimerExpiredSendNavy()` on destruction of all USSR and BadGuy forces. We don't want players sitting with nothing to do waiting on a timer to finish. I believe the OG stopped the timer and sent the ships in when the player destroyed everything before the timer finishes.
I'd also rename `TimerExpiredSendNavy()` to just `SendNavy()`.

Reply:
The logic is at 
mods/ra/maps/allies-11a/allies11a.lua
Line 73 and 161-179
It will count down until timer expired, when timer expired(Ticked==0), TimerExpiredSendNavy() will be activated.
When timer hasn't expired, if the players destroyed both USSR and BasGuy required units, it will set the timer to 0(Ticked==0), at the next tick period, TimerExpiredSendNavy() will be activated.
`TimerTicks = DateTime.Minutes(72)`
`Ticked = TimerTicks`
`Tick = function()`
`       USSR.Cash = 5000`
`	BadGuy.Cash = 10000`
`	if Greece.HasNoRequiredUnits() then`
`		USSR.MarkCompletedObjective(USSRObj)`
`	end`
`	if Ticked > 0 then`
`		UserInterface.SetMissionText("Naval vessels arrive in " .. Utils.FormatTime(Ticked), TimerColor)`
`		Ticked = Ticked - 1`
`		if USSR.HasNoRequiredUnits() and BadGuy.HasNoRequiredUnits() then`
`			Ticked = 0`
`		end`
`	elseif Ticked == 0 then`
`		FinishTimer()`
`		TimerExpiredSendNavy()`
`		Ticked = Ticked - 1`
`	end`
`end`
similar logic has been used at
mods/ra/maps/survival02/survival02.lua, line 79-128

Reply:
I see, I was looking for a second function. Your way looks cleaner.